device/core/src/blob_upload/blob_upload_result.ts

```diff
@@ -40,30 +40,40 @@ export class BlobUploadResult {
    *
    * @throws {ReferenceException} If err is null and either body or response are also falsy.
    */
+  // DISCUSS WITH PIERRE BECAUSE THESE BLOB REQUIREMENTS DONT MAKE SENSE AT ALL
```

Comment:
> DISCUSS WITH PIERRE BECAUSE THESE BLOB REQUIREMENTS DONT MAKE SENSE AT ALL [](start = 4, length = 75)

This ^^^^^^^^^^^^ really seems like something that should be done and deleted.

And as an aside, I like comments.  I would have written a bunch more comments in the below code.  If only for my own sanity while writing it.  It perhaps might have made more sense to put the requirements closer to the actual lines implementing them.